antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/RuleContext.kt

```diff
@@ -119,7 +119,7 @@ public open class RuleContext : RuleNode {
       return builder.toString()
     }
 
-  public open var ruleIndex: Int = -1
+  public open val ruleIndex: Int = -1
```

Comment:
Should we introduce a constant called `INVALID_RULE_INDEX` or `UNASSIGNED_RULE_INDEX` set to 1 and refer to that constant here and in the constructor of `InterpreterRuleContext`?

Reply:
Make sense. I would then consider adding a comment instead.
Perhaps we could add an issue about that in the antlr4 repo, but that is definitely out of the scope of this PR

Reply:
@ftomassetti addressed!

Reply:
Thank you!